storage/innobase/handler/ha_innodb.cc

```diff
@@ -8840,15 +8840,15 @@ ha_innobase::update_row(
 	                   && table->vers_start_id() == 0;
 
 	if (force_insert) {
-	  table->vers_start_field()->store(trx->id);
+		table->vers_start_field()->store(trx->id);
 
-	  ptrdiff_t offset= table->vers_start_field()->ptr - table->record[0];
-    longlong a= sint8korr(table->record[0] + offset);
-    longlong b= sint8korr(table->record[1] + offset);
+		Field *start= table->vers_start_field();
+		longlong old_val= uint8korr(start->ptr_in_record(old_row));
+		longlong new_val= uint8korr(start->ptr_in_record(new_row));
```

Comment:
Was it intentionally to ignore p.2?

Reply:
Sorry, my comment with arguments was accidentally deleted (guess i've pressed "cancel review")

Is there any reason to prefer `memcmp` over comparing two integers?

My reasons not to prefer are:
- integer comparison works faster
- cleaner to read
- `Field_longlong::cmp` does it in the same way (apparently, `Field_ulonglong` does not)